profiler/options.go

```diff
@@ -115,6 +116,15 @@ type config struct {
 
 // logStartup records the configuration to the configured logger in JSON format
 func logStartup(c *config) {
+	url := c.targetURL
+	if c.udsPath != "" {
+		// We don't actually use the UNIX path in our upload requests.
+		// The UNIX path is used in the underlying transport, so the
+		// client doesn't care what the host part of the URL is, as long
+		// as the path ("/profiling/v1/input") is right. Nonetheless, we
+		// should give some signal that UDS was configured properly.
+		url = fmt.Sprintf("unix://%s", c.udsPath)
```

Comment:
**TODO:** this was the quickest way to get from point A to point B. But we might also get confusing logs if an upload fails, and we see the URL in the path. I didn't touch the `targetURL` that's used in the upload, because in my experience the Go HTTP client still expects a proper request URL and it won't do the right thing if you try to shove a UDS path into the URL (where does the file path end and the resource path begin?). You can escape the UDS path, but IMO that just doesn't look right. This approach at leas preserves what the user will have provided more accurately.

Reply:
Tried it out, here's what I see:

```
2023/03/24 09:45:04 Datadog Tracer v1.49.0 ERROR: Failed to upload profile: failed after 2 retries, last error was: Post "http://localhost:8126/profiling/v1/input": dial unix /var/foo/bar: connect: no such file or directory, 27 additional messages skipped (first occurrence: 24 Mar 23 09:44 EDT)
```

So, we see that the upload was happening over UDS like expected, but we also still see the default `localhost:8126` address. Worth fixing more?

Reply:
We dealt with this in the tracer too. Here's how I fixed it there:
https://github.com/DataDog/dd-trace-go/pull/1604